### PR TITLE
add docs for enforce mpp

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add docs for enforce mpp

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6250
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -596,6 +599,16 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to control whether to enable the support for window functions. Note that window functions may use reserved keywords. This might cause SQL statements that could be executed normally cannot be parsed after upgrading TiDB. In this case, you can set `tidb_enable_window_function` to `OFF`.
 
+### `tidb_enforce_mpp` <span class="version-mark">New in v5.1</span>
+
+- Scope: SESSION
+- Default value: `OFF`
+- This variable is used to control whether to ignore the optimizer cost estimation, and to force the use of TiFlash's MPP mode to execute the queries. You can set the following values:
```
Comment (hanfei1991) [inline system-variables.md:606]: 1. comma "," is usually not used before "and" in a compound sentence.
2. "to execute the queries" is a little redundant to me. Besides, "force to do sth" is more natural to me. How about "force to use TiFlash MPP mode"?

I'm not an English expert. These are only personal suggestions.

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -439,6 +439,12 @@ Configuration items related to performance.
 + Default value: `20971520`
 + When `nested-loop-join-cache-capacity` is set to `0`, nested loop join cache is disabled by default. When the LRU size is larger than the value of `nested-loop-join-cache-capacity`, the elements in the LRU are removed.
 
+### `enforce-mpp`
+
++ Determines whether to ignore the optimizer cost estimation, and force TiFlash's MPP mode to execute queries.
```
Comment (hanfei1991) [inline tidb-configuration-file.md:444]: and to force?

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+For example, if you do not want to use MPP mode, you can execute the following statement:
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
+```
+
+If you want to intelligently choose whether to use the MPP mode (by default) using the cost estimation of TiDB optimizer, you can execute the following statement:
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:284]: This condition seems too long to read. How about replacing `using the cost est of TiDB optimizer` by 'by TiDB optimizer'? That makes sense to readers

File: tiflash/use-tiflash.md [hunk 4]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
+```
+
+If you want TiDB to ignore the optimizer's cost estimates and force to choose the MPP mode, you can execute the following statement:
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:293]: and to force?

Comment (hanfei1991) [reply tiflash/use-tiflash.md:293]: statments?

File: tiflash/use-tiflash.md [hunk 5]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=1;
 ```
 
+The initial value of the Session variable `tidb_enforce_mpp` is equal to the[`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration item value of this tidb-server instance (which is `false` by default). In a TiDB cluster, if several tidb-server instances only perform analytical queries, to ensure that they can choose MPP mode, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value to `true`.
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:302]: is "Session" need to capitalize the first letter?

Comment (hanfei1991) [reply tiflash/use-tiflash.md:302]: When I first read the sentence "In a TIDB cluster ...", I'm quite confused.  I think you want to express "If you want to make several tidb servers only serve analytical quries in the cluster and to force them to use MPP mode," I think this way is clearer.

File: tiflash/use-tiflash.md [hunk 6]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+> **Note:**
+>
+> When `tidb_enforce_mpp=1` takes effect, the TiDB optimizer will ignore the cost estimation to choose MPP mode. However, TiDB will not choose the MPP mode if other factors that do not support the MPP mode occur, such as no TiFlash copy, the replication of TiFlash copies is not completed, and statements contain operators or functions that are not supported by the MPP mode.
+> 
+> If the TiDB optimizer cannot select MPP mode due to reasons other than cost estimation, when you use the `EXPLAIN` statement to view the execution plan, a warning is returned to explain the reason, for example:
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:308]: sorry to be frank, this sentence is quite messy, so I think we have to reorganize the sentence. How about "When TiDB optimizer choose non-MPP plan unexpectly, we can use `explain` statement to check out the plan and the warnings which explain why it doesn't apply MPP mode". For example

**Comment by hanfei1991** [comment]
I think the use of "the" is not accurate in this document. We don't say "the TiFlash engine" or "the TiDB optimizer" because there is only one TiFlash engine and TiDB optimizer. I'm not English expert so that's only my intuition.

File: tiflash/use-tiflash.md [hunk 7]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=1;
 ```
 
+The initial value of the Session variable `tidb_enforce_mpp` is equal to the[`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration item value of this tidb-server instance (which is `false` by default). In a TiDB cluster, if several tidb-server instances only perform analytical queries, to ensure that they can choose MPP mode, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value to `true`.
+
+> **Note:**
+>
+> When `tidb_enforce_mpp=1` takes effect, the TiDB optimizer will ignore the cost estimation to choose MPP mode. However, TiDB will not choose the MPP mode if other factors that do not support the MPP mode occur, such as no TiFlash copy, the replication of TiFlash copies is not completed, and statements contain operators or functions that are not supported by the MPP mode.
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:306]: It's not natural to use "copy" to describe the follower role in a cluster. The most often used term is "replica". "if other factors that do not support the MPP mode occur" is a little awkward. If I wrote it, I would say "However, there are still some cases that block MPP mode, such as missing TiFlash replica for some tables ..."

File: system-variables.md [hunk 8]
```diff
@@ -596,6 +599,16 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to control whether to enable the support for window functions. Note that window functions may use reserved keywords. This might cause SQL statements that could be executed normally cannot be parsed after upgrading TiDB. In this case, you can set `tidb_enable_window_function` to `OFF`.
 
+### `tidb_enforce_mpp` <span class="version-mark">New in v5.1</span>
+
+- Scope: SESSION
+- Default value: `OFF`
+- This variable is used to control whether to ignore the optimizer cost estimation and force queries to be executed using TiFlash's MPP mode. You can set the following values:
+    - `0` or `OFF`, which means MPP mode is not mandatory (default).
```
Comment (LittleFall) [inline system-variables.md:607]: Maybe we can unify the expression of `default`? In `tidb_allow_mpp` it's `by default`, here is `default`.

File: tiflash/use-tiflash.md [hunk 9]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+If you want TiDB to ignore the optimizer's cost estimations and force to choose the MPP mode, you can execute the following statements:
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=1;
 ```
 
+The initial value of the session variable `tidb_enforce_mpp` is equal to the[`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration item value of this tidb-server instance (which is `false` by default). If you want to make several tidb servers only serve analytical quries in a TiDB cluster and to force them to use MPP mode, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value to `true`.
```
Comment (LittleFall) [inline tiflash/use-tiflash.md:302]:
```suggestion
The initial value of the session variable `tidb_enforce_mpp` is equal to the [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration item value of this tidb-server instance (which is `false` by default). If you want to make several tidb servers only serve analytical quries in a TiDB cluster and to force them to use MPP mode, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value to `true`.
```

**Review by LittleFall — COMMENTED** [review]
others lgtm

File: system-variables.md [hunk 10]
```diff
@@ -206,8 +206,11 @@ mysql> SELECT * FROM t1;
 
 - Scope: SESSION | GLOBAL
 - Default value: `ON`
-- Possible values: `OFF`, `ON`, `ENFORCE`
-- This variable controls whether to use the MPP mode of TiFlash to execute queries. If the value is set to `ON`, TiDB automatically determines using the optimizer whether to choose MPP to execute queries. MPP is a distributed computing framework provided by the TiFlash engine, which allows data exchange between nodes and provides high-performance, high-throughput SQL algorithms.
+- This variable controls whether to use the MPP mode of TiFlash to execute queries. You can set the following values, including:
+    - `0` or `OFF`, which means never use MPP mode.
```
Comment (TomShawn) [inline system-variables.md:210]:
```suggestion
    - `0` or `OFF`, which means that the MPP mode will not be used.
```

File: system-variables.md [hunk 11]
```diff
@@ -206,8 +206,11 @@ mysql> SELECT * FROM t1;
...
+    - `1` or `ON`, which means that the optimizer determines whether to use MPP mode based on the cost estimation (by default).
```
Comment (TomShawn) [inline system-variables.md:211]:
```suggestion
    - `1` or `ON`, which means that the optimizer determines whether to use the MPP mode based on the cost estimation (by default).
```

File: system-variables.md [hunk 12]
```diff
@@ -206,8 +206,11 @@ mysql> SELECT * FROM t1;
...
+
+MPP is a distributed computing framework provided by the TiFlash engine, which allows data exchange between nodes and provides high-performance, high-throughput SQL algorithms. For details about MPP mode selection, refer to [Control whether to choose the MPP mode](/tiflash/use-tiflash.md#control-whether-to-choose-the-mpp-mode).
```
Comment (TomShawn) [inline system-variables.md:213]:
```suggestion
MPP is a distributed computing framework provided by the TiFlash engine, which allows data exchange between nodes and provides high-performance, high-throughput SQL algorithms. For details about the selection of the MPP mode, refer to [Control whether to choose the MPP mode](/tiflash/use-tiflash.md#control-whether-to-choose-the-mpp-mode).
```

File: system-variables.md [hunk 13]
```diff
@@ -606,6 +609,16 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to control whether to enable the support for window functions. Note that window functions may use reserved keywords. This might cause SQL statements that could be executed normally cannot be parsed after upgrading TiDB. In this case, you can set `tidb_enable_window_function` to `OFF`.
 
+### `tidb_enforce_mpp` <span class="version-mark">New in v5.1</span>
+
+- Scope: SESSION
+- Default value: `OFF`
+- This variable is used to control whether to ignore the optimizer cost estimation and force queries to be executed using TiFlash's MPP mode. You can set the following values:
```
Comment (TomShawn) [inline system-variables.md:616]:
```suggestion
- Controls whether to ignore the optimizer's cost estimation and to force queries to be executed using TiFlash's MPP mode. The value options are as follows:
```

File: system-variables.md [hunk 14]
```diff
@@ -606,6 +609,16 @@ Query OK, 0 rows affected (0.09 sec)
...
+    - `0` or `OFF`, which means MPP mode is not mandatory (by default).
```
Comment (TomShawn) [inline system-variables.md:617]:
```suggestion
    - `0` or `OFF`, which means that the MPP mode is not forcibly used (by default).
```

File: system-variables.md [hunk 15]
```diff
@@ -606,6 +609,16 @@ Query OK, 0 rows affected (0.09 sec)
...
+    - `1` or `ON`, which means that the cost estimation is ignored and MPP mode is forced to use. Note that this setting only takes effect when `tidb_allow_mpp=true`.
```
Comment (TomShawn) [inline system-variables.md:618]:
```suggestion
    - `1` or `ON`, which means that the cost estimation is ignored and the MPP mode is forcibly used. Note that this setting only takes effect when `tidb_allow_mpp=true`.
```

File: system-variables.md [hunk 16]
```diff
@@ -606,6 +609,16 @@ Query OK, 0 rows affected (0.09 sec)
...
+
+MPP is a distributed computing framework provided by the TiFlash engine, which allows data exchange between nodes and provides high-performance, high-throughput SQL algorithms. For details about MPP mode selection, refer to [Control whether to choose the MPP mode](/tiflash/use-tiflash.md#control-whether-to-choose-the-mpp-mode).
```
Comment (TomShawn) [inline system-variables.md:620]:
```suggestion
MPP is a distributed computing framework provided by the TiFlash engine, which allows data exchange between nodes and provides high-performance, high-throughput SQL algorithms. For details about the selection of the MPP mode, refer to [Control whether to choose the MPP mode](/tiflash/use-tiflash.md#control-whether-to-choose-the-mpp-mode).
```

File: tidb-configuration-file.md [hunk 17]
```diff
@@ -439,6 +439,12 @@ Configuration items related to performance.
 + Default value: `20971520`
 + When `nested-loop-join-cache-capacity` is set to `0`, nested loop join cache is disabled by default. When the LRU size is larger than the value of `nested-loop-join-cache-capacity`, the elements in the LRU are removed.
 
+### `enforce-mpp`
+
++ Determines whether to ignore the optimizer cost estimation and force TiFlash's MPP mode to execute queries.
```
Comment (TomShawn) [inline tidb-configuration-file.md:444]:
```suggestion
+ Determines whether to ignore the optimizer's cost estimation and to force queries to be executed using TiFlash's MPP mode.
```

File: tiflash/use-tiflash.md [hunk 18]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
 ## Use the MPP mode
 
-TiFlash supports using the MPP mode to execute queries, which introduces cross-node data exchange (data shuffle process) into the computation. The MPP mode is enabled by default and can be disabled by setting the value of the global/session variable [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-new-in-v50) to `0` or `OFF`.
+TiFlash supports using the MPP mode to execute queries, which introduces cross-node data exchange (data shuffle process) into the computation. TiDB automatically determines whether to choose the MPP mode using the optimizer. You can change the selection strategy by modifying the values of [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-new-in-v50) and [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-new-in-v51).
 
-```shell
-set @@session.tidb_allow_mpp=0
+### Control whether to choose the MPP mode
+
+`tidb_allow_mpp` controls whether TiDB can select MPP mode to execute queries. `tidb_enforce_mpp` controls whether the optimizer cost estimate is ignored and the MPP mode of TiFlash is forced to execute queries.
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:266]:
```suggestion
The `tidb_allow_mpp` variable controls whether TiDB can select the MPP mode to execute queries. The `tidb_enforce_mpp` variable controls whether the optimizer's cost estimation is ignored and the MPP mode of TiFlash is forcibly used to execute queries.
```

File: tiflash/use-tiflash.md [hunk 19]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+The results corresponding to all values of these two variables are as follows:
+
+|                        | tidb_allow_mpp=off | tidb_allow_mpp=on (by default)              |
+| ---------------------- | -------------------- | -------------------------------- |
+| tidb_enforce_mpp=off (by default) | MPP mode is not used. | The optimizer selects the MPP mode based on cost estimation. (by default)|
+| tidb_enforce_mpp=on  | MPP mode is not used.   | TiDB ignores the cost estimate and chooses MPP mode.      |
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:273]:
```suggestion
| tidb_enforce_mpp=off (by default) | The MPP mode is not used. | The optimizer selects the MPP mode based on cost estimation. (by default)|
| tidb_enforce_mpp=on  | MPP mode is not used.   | TiDB ignores the cost estimation and chooses the MPP mode.      |
```

File: tiflash/use-tiflash.md [hunk 20]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+For example, if you do not want to use MPP mode, you can execute the following statements:
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
+```
+
+If you want to use the cost estimation of TiDB optimizer to intelligently choose whether to use MPP mode (by default), you can execute the following statements:
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:284]:
```suggestion
If you want TiDB to use the optimizer's cost estimation to intelligently choose whether to use the MPP mode (by default), you can execute the following statements:
```

File: tiflash/use-tiflash.md [hunk 21]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
+```
+
+If you want TiDB to ignore the optimizer's cost estimations and force to choose the MPP mode, you can execute the following statements:
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:293]:
```suggestion
If you want TiDB to ignore the optimizer's cost estimation and to forcibly choose the MPP mode, you can execute the following statements:
```

File: tiflash/use-tiflash.md [hunk 22]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=1;
 ```
 
+The initial value of the session variable `tidb_enforce_mpp` is equal to the [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration item value of this tidb-server instance (which is `false` by default). If you want to make several tidb servers only serve analytical quries in a TiDB cluster and to force them to use MPP mode, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value to `true`.
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:302]:
```suggestion
The initial value of the `tidb_enforce_mpp` session variable is equal to the [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value of this tidb-server instance (which is `false` by default). If multiple tidb-server instances in a TiDB cluster only perform analytical queries and you want to make sure that the MPP mode is used on these instances, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration values to `true`.
```

File: tiflash/use-tiflash.md [hunk 23]
```diff
@@ -259,12 +259,73 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+> **Note:**
+>
+> When `tidb_enforce_mpp=1` takes effect, TiDB optimizer will ignore the cost estimation to choose MPP mode. However, TiDB will not select MPP mode if other factors block MPP mode, such as no TiFlash replica, the replication of TiFlash replicas is not completed, and statements contain operators or functions that are not supported by MPP mode.
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:306]:
```suggestion
> When `tidb_enforce_mpp=1` takes effect, the TiDB optimizer will ignore the cost estimation to choose the MPP mode. However, if other factors block the MPP mode, TiDB will not select the MPP mode. These factors include the absence of TiFlash replica, unfinished replication of TiFlash replicas, and statements containing operators or functions that are not supported by MPP mode.
```

**Review by TomShawn — COMMENTED** [review]
+ 建议 Select 和 Choose 不要频繁交替使用。建议 PR 里统一用词。
+ 请给所有 MPP mode 统一加定冠词 the

**Comment by Joyinqin** [comment]
@TomShawn @hanfei1991 @LittleFall PTAL again, thx~

File: system-variables.md [hunk 24]
```diff
@@ -606,6 +609,16 @@ Query OK, 0 rows affected (0.09 sec)
 - Default value: `ON`
 - This variable is used to control whether to enable the support for window functions. Note that window functions may use reserved keywords. This might cause SQL statements that could be executed normally cannot be parsed after upgrading TiDB. In this case, you can set `tidb_enable_window_function` to `OFF`.
 
+### `tidb_enforce_mpp` <span class="version-mark">New in v5.1</span>
+
+- Scope: SESSION
+- Default value: `OFF`
+- Controls whether to ignore the optimizer's cost estimation and to force queries to be executed using TiFlash's MPP mode. The value options are as follows:
```
Comment (TomShawn) [inline system-variables.md:616]:
```suggestion
- Controls whether to ignore the optimizer's cost estimation and to forcibly use TiFlash's MPP mode for query execution. The value options are as follows:
```

File: tidb-configuration-file.md [hunk 25]
```diff
@@ -439,6 +439,12 @@ Configuration items related to performance.
 + Default value: `20971520`
 + When `nested-loop-join-cache-capacity` is set to `0`, nested loop join cache is disabled by default. When the LRU size is larger than the value of `nested-loop-join-cache-capacity`, the elements in the LRU are removed.
 
+### `enforce-mpp`
+
++ Determines whether to ignore the optimizer's cost estimation and to force queries to be executed using TiFlash's MPP mode.
```
Comment (TomShawn) [inline tidb-configuration-file.md:444]:
```suggestion
+ Determines whether to ignore the optimizer's cost estimation and to forcibly use TiFlash's MPP mode for query execution.
```

File: tiflash/use-tiflash.md [hunk 26]
```diff
@@ -259,13 +259,74 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+
+{{< copyable "sql" >}}
+
+```sql
+set @@session.tidb_allow_mpp=1;
+set @@session.tidb_enforce_mpp=0;
 ```
 
-MPP mode supports these physical algorithms: Broadcast Hash Join, Shuffled Hash Join, Shuffled Hash Aggregation, Union All, TopN, and Limit. The optimizer automatically determines which algorithm to be used in a query. To check the specific query execution plan, you can execute the `EXPLAIN` statement. If the result of the `EXPLAIN` statement shows ExchangeSender and ExchangeReceiver operators, it indicates that the MPP mode has taken effect.
+If you want TiDB to use the optimizer's cost estimation to intelligently select whether to use the MPP mode (by default), you can execute the following statements:
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:284]: @windtalker PTAL thanks!
```suggestion
If you want TiDB to consider the optimizer's cost estimation to intelligently select whether to use the MPP mode (by default), you can execute the following statements:
```

File: tiflash/use-tiflash.md [hunk 27]
```diff
@@ -259,13 +259,74 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
 
 ## Use the MPP mode
 
-TiFlash supports using the MPP mode to execute queries, which introduces cross-node data exchange (data shuffle process) into the computation. The MPP mode is enabled by default and can be disabled by setting the value of the global/session variable [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-new-in-v50) to `0` or `OFF`.
+TiFlash supports using the MPP mode to execute queries, which introduces cross-node data exchange (data shuffle process) into the computation. TiDB automatically determines whether to select the MPP mode using the optimizer. You can change the selection strategy by modifying the values of [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-new-in-v50) and [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-new-in-v51).
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:262]:
```suggestion
TiFlash supports using the MPP mode to execute queries, which introduces cross-node data exchange (data shuffle process) into the computation. TiDB automatically determines whether to select the MPP mode using the optimizer's cost estimation. You can change the selection strategy by modifying the values of [`tidb_allow_mpp`](/system-variables.md#tidb_allow_mpp-new-in-v50) and [`tidb_enforce_mpp`](/system-variables.md#tidb_enforce_mpp-new-in-v51).
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: tiflash/use-tiflash.md (hunk 26 above)
Comment (hanfei1991) [reply tiflash/use-tiflash.md:284]: How about "If you want TiDB's cost-based optimizer to automatically decide whether to use MPP mode.". Cost-based opimizer is a more common terms and makes sense to most readers.

File: tiflash/use-tiflash.md [hunk 28]
```diff
@@ -259,13 +259,74 @@ If a query encounters unsupported push-down calculations, TiDB needs to complete
...
+set @@session.tidb_enforce_mpp=1;
+```
+
+The initial value of the `tidb_enforce_mpp` session variable is equal to the [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration value of this tidb-server instance (which is `false` by default). If multiple tidb-server instances in a TiDB cluster only perform analytical queries and you want to make sure that the MPP mode is used on these instances, you can change their [`enforce-mpp`](/tidb-configuration-file.md#enforce-mpp) configuration values to `true`.
+
+> **Note:**
+>
+> When `tidb_enforce_mpp=1` takes effect, the TiDB optimizer will ignore the cost estimation to choose the MPP mode. However, if other factors block the MPP mode, TiDB will not select the MPP mode. These factors include the absence of TiFlash replica, unfinished replication of TiFlash replicas, and statements containing operators or functions that are not supported by the MPP mode.
+> 
+> If TiDB optimizer cannot select the MPP mode due to reasons other than cost estimation, when you use the `EXPLAIN` statement to view the execution plan, a warning is returned to explain the reason. For example:
```
Comment (hanfei1991) [inline tiflash/use-tiflash.md:308]: "to view" is unnatural. line 329 in this file uses 'check'. That makes more sense, such as "check out something".

File: tiflash/use-tiflash.md (hunk 26 above)
Comment (TomShawn) [reply tiflash/use-tiflash.md]: committed.

File: tiflash/use-tiflash.md (hunk 28 above)
Comment (TomShawn) [reply tiflash/use-tiflash.md:308]:
```suggestion
> If TiDB optimizer cannot select the MPP mode due to reasons other than cost estimation, when you use the `EXPLAIN` statement to check out the execution plan, a warning is returned to explain the reason. For example:
```

**Review by hanfei1991 — APPROVED** [review]

**Pull request merged**